examples/chapter1/annexa/src/example6.cpp

```diff
@@ -0,0 +1,48 @@
+// Copyright 2023 Klepsydra Technologies AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "simple_publisher.h"
+#include <iostream>
+
+#include <klepsydra/core/event_emitter_middleware_provider.h>
+int main()
+{
+    std::cout << "Main thread ID: " << std::this_thread::get_id() << std::endl;
+
+    kpsr::EventEmitterMiddlewareProvider<std::string> provider(nullptr,
+                                                               "tutorial_app_api_example6",
+                                                               0,
+                                                               nullptr,
+                                                               nullptr);
+
+    SimplePublisher simplePublisher(provider.getPublisher());
+    std::string captured_message = "Hello World! I am a captured message!";
+
+    provider.getSubscriber()
+        ->registerListener("example6", [&captured_message](const std::string &message) {
+            std::cout << "Message captured by the lambda: " << captured_message << std::endl;
+            captured_message = "Since I am reference captured in the lambda, my message is "
+                               "modified in the outside scope too!";
+            std::cout << "Message captured by the lambda: " << captured_message << std::endl;
+            std::cout << "Message received: " << message << std::endl;
+            std::cout << "Provider (subscriber) thread ID: " << std::this_thread::get_id()
+                      << std::endl;
+        });
+
+    simplePublisher.run();
```

Comment:
always publishes "Hello World!", unrelated with the `captured_message` var.
Maybe that is the intention but, alternatively, `run()` could take the event/message as an argument, as in [simple_event_publisher.h](https://github.com/klepsydra-technologies/kpsr-tutorial/blob/87121e41d881b344b7cf344798634c75966bdaea/examples/chapter1/annexb/include/simple_event_publisher.h#L32)

Reply:
The intention of these examples is to illustrate the different behaviours a lambda function can have, so they focus on the lambda used in the registerListener function instead of the subscriber/publisher flow. Passing the `captured_message` variable to the `run()` function would add an extra effort for the reader in my opinion.

Reply:
Agreed, no need to change the `run` in these examples.